openstates/me/people.py

```diff
@@ -31,67 +32,112 @@ def scrape(self, chamber=None):
             yield from self.scrape_reps()
 
     def scrape_reps(self):
-        url = 'http://www.maine.gov/legis/house/dist_mem.htm'
-        page = self.get(url).text
-        page = lxml.html.fromstring(page)
-        page.make_links_absolute(url)
-
-        # These do not include the non-voting tribal representatives
-        # They do not have numbered districts, and lack a good deal of
-        # the standard profile information about representatives
-        for district in page.xpath('//a[contains(@href, "dist_twn")]/..'):
-            if "- Vacant" in district.text_content():
-                self.warning("District is vacant: '{}'".
-                             format(district.text_content()))
+        mapping = {
+            'district': 0,
+            'last_name': 1,
+            'first_name': 2,
+            'mid_name': 3,
+            'jr_sr': 4,
+            'party': 5,
+            'legal_res': 6,
+            'address1': 7,
+            'address2': 8,
+            'town': 9,
+            'state': 10,
+            'zip_code': 11,
+            'zip_plus4': 12,
+            'home_phone': 13,
+            'cell_phone': 14,
+            'business_phone': 15,
+            'other_phone': 16,
+            'fax': 17,
+            'email': 18,
+            'seat_no': 19,
+            'house_leadership': 20,
+            'committees': 21
+        }
+
+        url = 'http://legislature.maine.gov/house/' + str(self.latest_session()) + 'house.xlsx'
+        fn, result = self.urlretrieve(url)
+
+        wb = xlrd.open_workbook(fn)
+        sh = wb.sheet_by_index(0)
+
+        LEGISLATOR_ROSTER_URL = \
+            'http://legislature.maine.gov/house/dist_mem.htm'
+        roster_doc = lxml.html.fromstring(self.get(LEGISLATOR_ROSTER_URL).text)
+        roster_doc.make_links_absolute(LEGISLATOR_ROSTER_URL)
+
+        for rownum in range(1, sh.nrows):
+            # get fields out of mapping
+            d = {}
+            for field, col_num in mapping.items():
+                try:
+                    d[field] = str(sh.cell(rownum, col_num).value).strip()
+                except UnicodeEncodeError:
+                    # str typecasting will raise error for non-ascii characters.
+                    d[field] = sh.cell(rownum, col_num).value.encode('utf-8').strip()
+
+            try:
+                # xlrd reads this value as a float. However we save it as a string.
+                district_number = str(int(float(d['district'])))
+            except ValueError:
+                # Will reach here in case district number is blank. We will skip this row.
```

Comment:
Can you add a log here, to let the user know that we're skipping the row? Probably a `self.info`, since it's "expected" that we'd skip non-voting members.